### ISSUE TITLE
Section with Markdown shortcode inside it excludes slides from the resulting section

### ISSUE THREAD

**Issue by hbobenicio** [dzello/reveal-hugo#32]
If you put a Markdown shortcode inside a Section, slides after the markdown close tag doen't  belong to the current Section.

Example:

```
{{% section %}}

# My Section: Slide 1

Here the section begins...

---

{{< markdown >}}

# My Section: Slide 2

## I'm a mardown slide, but with custom HTML

<section data-noprocess>
  <h1>Hi, Reveal JS!</h1>
<section>

{{< /markdown >}}

---

# My Section (?): Slide 3

* this is not
* in the
* section any more :(

{{% /section %}}

```

Am I doing something wrong here? Or this issue is just there?

**Comment by dzello** [comment]
Thanks for reporting @hbobenicio. This could be a bug. Does it make a difference if you use the `{{<` tags for the section instead of the `{{%`?

**Comment by hbobenicio** [comment]
@dzello Thanks for the quick feedback.

you say like this:

```
{{< section >}}

some slides

{{< markdown >}}

some content

{{< /markdown >}}

some (unfortunately) slides that stays out of section somehow

{{</ section >}}

```

If that is the case, the same problem happens.

**Comment by hbobenicio** [comment]
Renamed the issue's title to better describe what's happening.

**Comment by dzello** [comment]
Thanks for clarifying the title, I will investigate to see what markup is being generated incorrectly.

Curious - is there a reason you're using the Markdown shortcode vs. markdown directly?

**Comment by hbobenicio** [comment]
I'm using the markdown shortcode just to define HTML slides (with `<section data-noprocess>`) in order to put some fontawesome icons in it.

I could also download them manually (or via npm) and serve them as static files... but I thought that could be a bit more complex solution to manage instead of plain a html slide.

**Comment by dzello** [comment]
Got it. Check out https://reveal-hugo.dzello.com/#/43. You should be able
to put in that HTML without the markdown shortcode wrapping it. Let me know
if that works.

**Comment by hbobenicio** [comment]
My bad after all! Thanks for the quick support!

I've already tried that approach **but** i was doing this:

```
<section data-noprocess>
  <h1>Hi, Reveal JS!</h1>
<section> <!-- whoops -->
```

instead of the obvious

```
# My Section: Slide 2

## I'm a mardown slide, but with custom HTML

<section data-noprocess>
  <h1>Hi, Reveal JS!</h1>
</section>  <!-- let's close it correctly now :P -->
```

which motivated me to use markdown shortcode in first place.
Anyway, typo fixed and no markdown shortcodes after all!

Really appreciate your help @dzello ! Thanks!

**Comment by dzello** [comment]
Ah hah! Glad that works :)

**Comment by hbobenicio** [comment]
^^'

Sorry for that but in my excitement about rendering HMTL I forgot to double check about slides in the section.

no markdown shortcode anymore but the issue still persists.

```
{{< section >}}

some slides here

---
<section data-noprocess>
  <h1>No markdown shortcode, but after this slides still get out of the section</h1>
</section>
---

some slides that still get out of the section somehow

{{</ section >}}

```

Better to keep this issue, rename it or open it with another name?

**Comment by dzello** [comment]
Does it work if you put at least one newline both before and after the `---`?

**Comment by dzello** [comment]
Also, can you verify you're using the latest version of Hugo and of reveal-hugo? There was recently a section-related fix.

**Comment by hbobenicio** [comment]
> Does it work if you put at least one newline both before and after the `---`?

Nope... my slides, section shortode and section html tag are all surrounded by newlines.

> Also, can you verify you're using the latest version of Hugo and of reveal-hugo? There was recently a section-related fix.

* upgraded hugo from `v0.54.0` to `v0.55.6`
* reveal-hugo already on the latest version

slides goes off the section still.

**Comment by dzello** [comment]
Sorry I mean two newlines.

Le lun. 20 mai 2019 à 19:42, Hugo Benício <notifications@github.com> a
écrit :

**Comment by hbobenicio** [comment]
Can this be related to reveal js itself? Maybe trying to reproduce it with plain html and reveaj js can help us here too. I can create a sandbox I think.

**Comment by hbobenicio** [comment]
two newlines didn't make it.
